docs/developer-guide/running-substreams.md

```diff
@@ -77,3 +77,36 @@ map_transfers: message "eth.erc721.v1.Transfers": {
 ```
 
 The example output shown above contains data for different transfers from data in the blockchain. These transfers can also be verified on [Etherscan](https://etherscan.io/tx/0xcfb197f62ec5c7f0e71a11ec0c4a0e394a3aa41db5386e85526f86c84b3f2796).
+
+## Development & Production Mode
+
+### Development Mode
+
+Production and development modes impact the execution of Substreams. Key aspects of execution include:
+
+- The time required to reach the first byte.
+- The speed that large ranges get executed.
+- The module logs and outputs that are sent back to the client.
+
+Differences between production and development modes include:
+
+- Forward execution is enabled in production mode and disabled in development mode.
+- The time required to reach the first byte in development mode is faster than in production mode.
+- In development mode, the client will receive all of the executed module's logs. In production mode the client will only receive the output module's logs.
+- In development mode, it's possible to request specific store snapshots in the execution tree.
+
+Examples: (given the dependencies: `[block] --> [map_pools] --> [store_pools] --> [map_transfers])`
+
+- Running the `substreams run substreams.yaml map_transfers` command will only print the `map_transfers` module's outputs and logs.
```

Comment:
It's not clear in the list below if running in production mode or development mode. I would introduce that fact the development is enabled by default in the CLI unless `-p` is specified. Also, a note below that should be added to emphasis the fact that we expect people to run the production mode using a Substreams `sink`. Very very rarely people should use `-p` as `substreams` CLI is really more a development tool than a production tool.

Reply:
Verbiage added to address the -p flag and development mode versus production mode. The changes will be available in the next commit/push.

Reply:
Let's add on each of them (list bullet points) that is runs in development mode.
```suggestion
- Running the `substreams run substreams.yaml map_transfers` command run in development mode and only print the `map_transfers` module's outputs and logs.
```

Something like this.

Reply:
Updated to reflect the content you've provided. Thank you. Changes will be available for review in the next commit/push.